### ISSUE TITLE
Source->Design selection is broken. Again.

### ISSUE THREAD

**Issue by peller** [maqetta/maqetta#3690]
In split view, click in source view within a widget's HTML.  The widget is not selected in design view.

**Comment by peller** [comment]
see #788, perhaps others.  Wonder if this has to do with clearing the undo stack #3074?

**Comment by JonFerraiolo** [comment]
I don't see how the fix for #3074 would impact the interaction between source view and design view.

**Comment by JonFerraiolo** [comment]
I took a second to put a breakpoint in Context.js:select() to see if the code ever gets to that routine when you click on the source code corresponding to a widget, and the debugger never gets to the breakpoint.

**Comment by peller** [comment]
Well, I did a search back in git, and it did indeed break with this checkin: 6108f37f3d0d1f5f2be884c554ffd300759bfd1f  However, if I revert the change, it doesn't work! 

My best guess was that something with the command stack triggered a clearing of the selection (though I don't think clear does that)

**Comment by JonFerraiolo** [comment]
Here is the full source code for the `clear` function in CommandStack.js:

```
    clear: function(){
        // summary:
        //      Clears the undo and redo stacks.
        this._undoStack = [];
        this._redoStack = [];
    },
```

Hard to conceive how calling `clear` could impact selection!

**Comment by peller** [comment]
@JonFerraiolo there's always that connect code, you know...  execute triggers all sorts of things.  clear doesn't, though perhaps it ought to.

**Comment by peller** [comment]
reopen for M10.  Need to merge davinciRuntime branch

**Comment by peller** [comment]
davinciRuntimeGlobal branch merged in 28763243f3466b859495e2e47889634a4a7b4488
